src/python/deepgnn/graph_engine/snark/convert.py

```diff
@@ -118,17 +118,31 @@ def convert(self):
                 num_workers=self.worker_count,
             )
 
-        for _, data in enumerate(self.file_iterator):
-            for line in data:
-                d.dispatch(line)
-
-        d.join()
+        if self.worker_count >= 1:
+            d = self.dispatcher
```

Comment:
You don't need a dispatcher in a single thread. Directly use writer.

Reply:
worker_count = 1 means that the TextFileIterator uses N workers but the dispatcher will actually use n_partitions workers so it does need the dispatcher.